common/src/main/scala/org/apache/comet/vector/NativeUtil.scala

```diff
@@ -56,6 +60,35 @@ class NativeUtil {
    */
   private val dictionaryProvider: CDataDictionaryProvider = new CDataDictionaryProvider
 
+  /**
+   * Allocates Arrow structs for the given number of columns.
+   *
+   * @param numCols
+   *   the number of columns
+   * @return
+   *   a pair of Arrow arrays and Arrow schemas
+   */
+  def allocateArrowStructs(numCols: Int): (Array[ArrowArray], Array[ArrowSchema]) = {
+    val arrays = new Array[ArrowArray](numCols)
+    val schemas = new Array[ArrowSchema](numCols)
+
+    (0 until numCols).foreach { index =>
+      val arrowSchema = ArrowSchema.allocateNew(allocator)
+
+      // Manually fill NULL to `release` slot of ArrowSchema because ArrowSchema doesn't provide
+      // `markReleased`.
+      val buffer =
+        MemoryUtil.directBuffer(arrowSchema.memoryAddress(), 72).order(ByteOrder.nativeOrder)
+      buffer.putLong(56, NULL);
```

Comment:
Could you add a comment explaining what 72 and 56 represent here?

Reply:
Added a comment.

Reply:
I'm OK with the current approach, but I still suggest a more clean way of setting base structure fields using snapshot:

```
ArrowSchema.Snapshot snapshot = arrowSchema.snapshot();
snapshot.release = NULL;
arrowSchema.save(snapshot);
```

Or use `std::ptr::write` on the rust side to avoid calling the garbled release handle completely.

Reply:
Okay. Let me change to use `std::ptr::write`.